// @sbpro/ng/dist/node-crud/files-internal/__plural@dasherize__/_shared-types-testing-__plural@dasherize__.ts
import {
    Create<%= classify(name) %>Payload,
    Delete<%= classify(name) %>Params,
    Read<%= classify(name) %>Params,
    Results<%= classify(name) %>,
    Update<%= classify(name) %>Params,
    Update<%= classify(name) %>Payload,
    DeleteMultiple<%= classify(plural) %>Payload,
} from '../../typings/<%= dasherize(plural) %>';

export class TestResults<%= classify(name) %> implements Results<%= classify(name) %> {
    id = '00000000-0000-0000-0000-0test<%= camelize(name) %>';
    displayName = 'DISPLAY_NAME';
    orgID = '00000000-0000-0000-0000-00000testorg';
}

/* Create */

export class TestCreate<%= classify(name) %>Payload implements Create<%= classify(name) %>Payload {
    displayName = 'DISPLAY_NAME';
}

/* Read */

export class TestRead<%= classify(name) %>Params implements Read<%= classify(name) %>Params {
    id = '00000000-0000-0000-0000-0test<%= camelize(name) %>';
}

export class TestReadMultiple<%= classify(plural) %>Payload implements ReadMultiple<%= classify(plural) %>Payload {
    <%= camelize(name) %>IDs = ['00000000-0000-0000-0000-0test<%= camelize(name) %>'];
}

/* Update */

export class TestUpdate<%= classify(name) %>Params implements Update<%= classify(name) %>Params {
    id = '00000000-0000-0000-0000-0test<%= camelize(name) %>';
}

export class TestUpdate<%= classify(name) %>Payload implements Update<%= classify(name) %>Payload {
    displayName = 'DISPLAY_NAME';
}

/* Delete */

export class TestDelete<%= classify(name) %>Params implements Delete<%= classify(name) %>Params {
    id = '00000000-0000-0000-0000-0test<%= camelize(name) %>';
}

export class TestDeleteMultiple<%= classify(plural) %>Payload implements DeleteMultiple<%= classify(plural) %>Payload {
    <%= camelize(name) %>IDs = ['00000000-0000-0000-0000-test<%= camelize(name) %>0', '00000000-0000-0000-0000-test<%= camelize(name) %>1'];
}

/* Find */

export class TestFind<%= classify(plural) %>Payload implements Find<%= classify(plural) %>Payload {
    select = ['SELECT'];
    relations = ['RELATIONS'];
}
