// @start-bootstrap/sb-clean-blog-shared-types/dist/testing/objects/post.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class TestResultsPost {
    constructor() {
        this.id = '00000000-0000-0000-0000-000000000001';
        this.slug = 'TEST_SLUG';
        this.backgroundImage = 'TEST_BACKGROUND_IMAGE';
        this.heading = 'TEST_HEADING';
        this.subHeading = 'TEST_SUB_HEADING';
        this.meta = 'TEST_META';
        this.body = 'TEST_BODY';
    }
}
exports.TestResultsPost = TestResultsPost;
/* Create */
class TestCreatePostPayload {
    constructor() {
        this.slug = 'TEST_SLUG';
        this.backgroundImage = 'TEST_BACKGROUND_IMAGE';
        this.heading = 'TEST_HEADING';
        this.subHeading = 'TEST_SUB_HEADING';
        this.meta = 'TEST_META';
        this.body = 'TEST_BODY';
    }
}
exports.TestCreatePostPayload = TestCreatePostPayload;
/* Read */
class TestReadPostParams {
    constructor() {
        this.id = '00000000-0000-0000-0000-000000000001';
    }
}
exports.TestReadPostParams = TestReadPostParams;
class TestReadPostQuery {
    constructor() {
        this.findBy = 'slug';
    }
}
exports.TestReadPostQuery = TestReadPostQuery;
/* Update */
class TestUpdatePostParams {
    constructor() {
        this.id = '00000000-0000-0000-0000-000000000001';
    }
}
exports.TestUpdatePostParams = TestUpdatePostParams;
class TestUpdatePostPayload {
    constructor() {
        this.backgroundImage = 'TEST_BACKGROUND_IMAGE';
        this.heading = 'TEST_HEADING';
        this.subHeading = 'TEST_SUB_HEADING';
        this.meta = 'TEST_META';
        this.body = 'TEST_BODY';
    }
}
exports.TestUpdatePostPayload = TestUpdatePostPayload;
/* Delete */
class TestDeletePostParams {
    constructor() {
        this.id = '00000000-0000-0000-0000-000000000001';
    }
}
exports.TestDeletePostParams = TestDeletePostParams;
