// @sbpro/ng/dist/node-crud/files-org/__plural@dasherize__/__plural@dasherize__-read.ts
import { find<%= classify(name) %> } from '@app/routes/_shared';
import { RouteAuthConfig } from '@lib/orm/models';
import { RouteHandlerMethodForConfig } from '@src/augmentations/fastify';
import { IDAndOrgIDParams, Results<%= classify(name) %>, roleName } from '@start-bootstrap/website-shared-types';
import { FastifyPlugin } from 'fastify';

import * as schemas from './_schemas';

export const <%= camelize(plural) %>Read: FastifyPlugin = async function (instance, options): Promise<void> {
    instance.route({
        method: 'GET',
        url: '/:id',
        schema: schemas.<%= camelize(plural) %>Read,
        handler,
        config: <RouteAuthConfig>{ minRole: roleName.editor },
    });
};

export const handler: RouteHandlerMethodForConfig<{
    Params: IDAndOrgIDParams;
}> = async function (request, reply): Promise<Results<%= classify(name) %>> {
    const read<%= classify(name) %>Params: IDAndOrgIDParams = request.params;
    const found<%= classify(name) %> = await find<%= classify(name) %>(request, read<%= classify(name) %>Params.id, {
        where: {
            organizationId: request.params.orgID,
        },
    });

    return found<%= classify(name) %>.toResults<%= classify(name) %>();
};
