// @sbpro/ng/dist/node-crud/files-org/__plural@dasherize__/__plural@dasherize__-all.ts
import { find<%= classify(plural) %>Filter } from '@app/routes/_shared';
import { RouteAuthConfig } from '@lib/orm/models';
import { RouteHandlerMethodForConfig } from '@src/augmentations/fastify';
import { OrgIDParam, Results<%= classify(name) %>, roleName } from '@start-bootstrap/website-shared-types';
import { FastifyPlugin } from 'fastify';

import * as schemas from './_schemas';

export const <%= camelize(plural) %>All: FastifyPlugin = async function (instance, options): Promise<void> {
    instance.route({
        method: 'GET',
        url: '/',
        schema: schemas.<%= camelize(plural) %>All,
        handler,
        config: <RouteAuthConfig>{ minRole: roleName.editor },
    });
};

export const handler: RouteHandlerMethodForConfig<{
    Params: OrgIDParam;
}> = async function (request, reply): Promise<Results<%= classify(name) %>[]> {
    const found<%= classify(plural) %> = await find<%= classify(plural) %>Filter(request, {
        where: {
            organizationId: request.params.orgID,
        },
    });

    return found<%= classify(plural) %>.map((<%= camelize(name) %>) => <%= camelize(name) %>.toResults<%= classify(name) %>());
};
