// @sbpro/ng/dist/node-crud/files-org/__plural@dasherize__/_schemas/schemas/__plural@dasherize__-body-find.schema.ts
export const <%= camelize(plural) %>BodyFindSchema = {
    type: 'object',
    properties: {
        select: {
            type: 'array',
            items: {
                type: 'string',
            },
        },
        relations: {
            type: 'array',
            items: {
                type: 'string',
            },
        },
        where: {
            type: 'array',
            items: {
                anyOf: [
                    {
                        type: 'object',
                        additionalProperties: {
                            type: ['string', 'number'],
                        },
                    },
                    {
                        type: 'object',
                        additionalProperties: {
                            type: 'object',
                            additionalProperties: {
                                type: ['string', 'number'],
                            },
                        },
                    },
                ],
            },
        },
        skip: { type: 'number' },
        take: { type: 'number' },
    },
    required: [],
};
