// @sbpro/ng/dist/node-routes/files-routes/__name@dasherize__/__name@dasherize__-action.ts
import * as schemas from '@app/routes/<%= dasherize(name) %>/_schemas';
import { RouteAuthConfig } from '@lib/orm/models';
import { RouteHandlerMethodForConfig } from '@src/augmentations/fastify';
import {
    <%= classify(name) %>ActionPayload,
    <%= classify(name) %>ActionResponse,
    roleName,
} from '@start-bootstrap/website-shared-types';
import { FastifyPluginAsync } from 'fastify';

export const <%= camelize(name) %>Action: FastifyPluginAsync = async function (instance, options): Promise<void> {
    instance.route({
        method: 'POST',
        url: '/start',
        schema: schemas.<%= camelize(name) %>Action,
        handler,
        config: <RouteAuthConfig>{ minRole: roleName.admin },
    });
};

export const handler: RouteHandlerMethodForConfig<{
    Body: <%= classify(name) %>ActionPayload;
}> = async function (request, reply): Promise<<%= classify(name) %>ActionResponse> {
    const <%= camelize(name) %>ActionPayload: <%= classify(name) %>ActionPayload = request.body;

    return {} as <%= classify(name) %>ActionResponse;
};
